bincode_derive's `Encode` and `Decode` proc macros fail if module declares `use anyhow::Ok`

Hi, first of all congratulations on the launch of bincode v2, I really like the changes.

During the migration to v2, I noticed the following minor issue, and I suspect that this is not intended behaviour:

```rs
use anyhow::Ok;

#[derive(Debug, PartialEq, bincode::Encode, bincode::Decode)]
struct Foo(String);

fn main() {
    let val = Foo("Bar".to_string());
    let config = bincode::config::standard();

    let val_encoded = bincode::encode_to_vec(&val, config).unwrap();
    let (val_decoded, _) = bincode::decode_from_slice(&val_encoded, config).unwrap();

    assert_eq!(val, val_decoded);
}
```

## Observed

Compiling the project fails because the derived `Encode` and `Decode` implementations return `Ok`, which is bound to `anyhow::Ok` due to the use declaration in L1.

<details>
<summary>Error</summary>

```
error[E0308]: mismatched types
 --> src/main.rs:3:28
  |
3 | #[derive(Debug, PartialEq, bincode::Encode, bincode::Decode)]
  |                            ^^^^^^^^^^^^^^^
  |                            |
  |                            expected `Result<(), EncodeError>`, found `Result<(), Error>`
  |                            expected `Result<(), EncodeError>` because of return type
  |
  = note: expected enum `Result<_, EncodeError>`
             found enum `Result<_, anyhow::Error>`
  = note: this error originates in the derive macro `bincode::Encode` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
 --> src/main.rs:3:45
  |
3 | #[derive(Debug, PartialEq, bincode::Encode, bincode::Decode)]
  |                                             ^^^^^^^^^^^^^^^
  |                                             |
  |                                             expected `Result<Foo, DecodeError>`, found `Result<Foo, Error>`
  |                                             expected `Result<Foo, DecodeError>` because of return type
  |
  = note: expected enum `Result<_, DecodeError>`
             found enum `Result<_, anyhow::Error>`
  = note: this error originates in the derive macro `bincode::Decode` (in Nightly builds, run with -Z macro-backtrace for more info)
```
</details>

## Expected

The project should compile.

--- a/derive/src/derive_enum.rs
+++ b/derive/src/derive_enum.rs
@@ -124,7 +124,7 @@ impl DeriveEnum {
                                     }
                                 }
                             }
-                            body.push_parsed("Ok(())")?;
+                            body.push_parsed("core::result::Result::Ok(())")?;
                             Ok(())
                         })?;
                         match_body.punct(',');
@@ -274,7 +274,7 @@ impl DeriveEnum {
                                 variant_case.push(variant_index.remove(0));
                             }
                             variant_case.puncts("=>");
-                            variant_case.ident_str("Ok");
+                            variant_case.push_parsed("core::result::Result::Ok")?;
                             variant_case.group(Delimiter::Parenthesis, |variant_case_body| {
                                 // Self::Variant { }
                                 // Self::Variant { 0: ..., 1: ... 2: ... },
@@ -384,7 +384,7 @@ impl DeriveEnum {
                                 variant_case.push(variant_index.remove(0));
                             }
                             variant_case.puncts("=>");
-                            variant_case.ident_str("Ok");
+                            variant_case.push_parsed("core::result::Result::Ok")?;
                             variant_case.group(Delimiter::Parenthesis, |variant_case_body| {
                                 // Self::Variant { }
                                 // Self::Variant { 0: ..., 1: ... 2: ... },

--- a/derive/src/derive_struct.rs
+++ b/derive/src/derive_struct.rs
@@ -56,7 +56,7 @@ impl DeriveStruct {
                         }
                     }
                 }
-                fn_body.push_parsed("Ok(())")?;
+                fn_body.push_parsed("core::result::Result::Ok(())")?;
                 Ok(())
             })?;
         Ok(())
@@ -95,7 +95,7 @@ impl DeriveStruct {
             .with_return_type(format!("core::result::Result<Self, {}::error::DecodeError>", crate_name))
             .body(|fn_body| {
                 // Ok(Self {
-                fn_body.ident_str("Ok");
+                fn_body.push_parsed("core::result::Result::Ok")?;
                 fn_body.group(Delimiter::Parenthesis, |ok_group| {
                     ok_group.ident_str("Self");
                     ok_group.group(Delimiter::Brace, |struct_body| {
@@ -174,7 +174,7 @@ impl DeriveStruct {
             .with_return_type(format!("core::result::Result<Self, {}::error::DecodeError>", crate_name))
             .body(|fn_body| {
                 // Ok(Self {
-                fn_body.ident_str("Ok");
+                fn_body.push_parsed("core::result::Result::Ok")?;
                 fn_body.group(Delimiter::Parenthesis, |ok_group| {
                     ok_group.ident_str("Self");
                     ok_group.group(Delimiter::Brace, |struct_body| {

--- a/tests/derive.rs
+++ b/tests/derive.rs
@@ -403,6 +403,23 @@ fn test_enum_with_generics_roundtrip() {
     assert_eq!(start, decoded);
 }
 
+mod derive_with_polluted_scope {
+    #[allow(dead_code)]
+    #[allow(non_snake_case)]
+    fn Ok() {}
+
+    #[derive(bincode::Encode, bincode::Decode)]
+    struct A {
+        a: u32,
+    }
+
+    #[derive(bincode::Encode, bincode::Decode)]
+    enum B {
+        A,
+        B,
+    }
+}
+
 #[cfg(feature = "alloc")]
 mod zoxide {
     extern crate alloc;

